server/services/chat/olly_chat_service.ts

```diff
@@ -58,40 +48,22 @@ export class OllyChatService implements ChatService {
       OllyChatService.abortControllers.set(payload.memory_id, new AbortController());
     }
 
-    // if rootAgentId has not been initialized yet, init rootAgentId firstly
-    if (!OllyChatService.rootAgentId) {
-      await this.initRootAgent();
-      this.context.assistant_plugin.logger.info(
-        `root agent id has not been initialized yet, init it at the first time, current root agent id is:${OllyChatService.rootAgentId}`
-      );
-    }
+    const rootAgentId = await this.getRootAgent();
 
     try {
-      return await this.callExecuteAgentAPI(payload);
+      return await this.callExecuteAgentAPI(payload, rootAgentId);
     } catch (error) {
-      if (
-        error.meta?.statusCode === RESOURCE_NOT_FOUND_STATUS_CODE &&
-        error.body.error.type === RESOURCE_NOT_FOUND_ERROR
-      ) {
-        const oldRootAgentId = OllyChatService.rootAgentId;
-        await this.initRootAgent();
-        this.context.assistant_plugin.logger.info(
-          `cannot find the root agent id: ${oldRootAgentId}, try to fetch the new root agent id, now it is:${OllyChatService.rootAgentId}`
-        );
-        return await this.callExecuteAgentAPI(payload);
-      } else {
-        throw error;
-      }
+      throw error;
```

Comment:
since it's just `throw error`, maybe remove the try catch?

Reply:
Yeah, I've changed that, thanks!